include/envoy/event/dispatcher.h

```diff
@@ -153,6 +153,17 @@ class Dispatcher {
    */
   enum class RunType { Block, NonBlock };
   virtual void run(RunType type) PURE;
+
+  /**
+   * Optionally sets a custom buffer factory for the dispatcher.  This factory may be used by new
+   * connections on connection creation.
+   */
+  virtual void setBufferFactory(Buffer::FactoryPtr&& factory) PURE;
```

Comment:
This would allow changing the factory to one with different behavior at runtime, but I'm not sure if that would ever be useful.